pennylane/math/single_dispatch.py

```diff
@@ -48,6 +48,7 @@ def _i(name):
 # qml.SparseHamiltonian are not automatically 'unwrapped' to dense NumPy arrays.
 ar.register_function("scipy", "to_numpy", lambda x: x)
 ar.register_function("scipy", "shape", np.shape)
+ar.register_function("scipy", "ndim", np.ndim)

```

Comment:
This is for sparse matrix support?

Reply:
We use `qml.math.ndim` and `scipy.ndim` raises a deprecation warning, so I mimicked the alias for `shape`, which also is deprecated in `scipy`.